src/zarr.hh

```diff
@@ -98,18 +101,26 @@ struct Zarr final : StorageInterface
     uint64_t max_bytes_per_chunk_;
     ImageShape image_shape_;
     TileShape tile_shape_;
-    std::vector<std::shared_ptr<ChunkWriter>> writers_;
-    size_t tiles_per_chunk_;
+    bool enable_multiscale_;
+
+    /// Downsampling of incoming frames.
+    std::optional<FrameScaler> frame_scaler_;
+
+    /// Chunk writers for each layer/scale
+    std::map<size_t, std::vector<std::shared_ptr<ChunkWriter>>> writers_;
 
     // changes during acquisition
-    size_t frame_count_;
+    uint32_t frame_count_;
```

Comment:
Since the number of frames written depends on the scale, the frame count is computed in each chunk writer. The only thing we really do with it is check that it's the first frame.